### PR TITLE
Code was improved via checkstyle

### LINKED ISSUE #13: Add checkstyle analyzer and improve code

### PR DESCRIPTION
**Pull request by vpetrovaa**
## PR-Codex overview
This PR adds new functionality to the NoteService application, including Kafka integration, Feign client, and Checkstyle configuration.

### Detailed summary
- Added Kafka support with `KfProducer` interface and implementation class `KfProducerImpl`.
- Created `KfProperties` class to hold Kafka configuration properties.
- Added `KfProducerConfig` class to configure Kafka producer.
- Added `NoteEvent` class to represent Kafka messages.
- Added `ElasticClient` interface to integrate with Elasticsearch using Feign client.
- Added `ConstraintsUtils` class to hold validation constraints.
- Configured Checkstyle rules and added suppressions file.

> The following files were skipped due to too many changes: `src/main/java/com/solvd/noteservice/web/controller/NoteController.java`, `src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/main/java/com/solvd/noteservice/kafka/KfProducer.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.solvd.noteservice.kafka;
+
+import com.solvd.noteservice.kafka.event.NoteEvent;
+
+public interface KfProducer {
+
+    void sendMessage(NoteEvent event);
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/kafka/KfProducer.java:7]: @vpetrovaa This interface absolutely not scalable

assuming, that tomorrow you will have another data objects to be sent to Kafka, etc

so, you will be needed to add another method to this interface, that originally needs only a method like `send(data)`

so, please fix this

File: src/main/java/com/solvd/noteservice/kafka/KfProducerImpl.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.solvd.noteservice.kafka.property.KfProperties;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class KfProducerImpl implements KfProducer {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/kafka/KfProducerImpl.java:13]: produce more reusable code, in the review of `interface KfProducer` I explain what is the problem with this code

File: src/main/java/com/solvd/noteservice/kafka/event/NoteEvent.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    private Method type;
+    private Long id;
+    private String description;
+    private String theme;
+    private String tag;
+    private Long userId;
+
+    public enum Method {
+        PUT, DELETE, POST;
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/kafka/event/NoteEvent.java:20]: I think `POST` is connected to HTTP, but your object is not connected (at least directly) to anything HTTP aware of

So, I suggest refactoring this `POST` constant's name to `SAVE`

File: src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java [hunk 4]
```diff
@@ -17,55 +20,86 @@
...
+        return elasticClient.isExistById(id);
     }
 
     @Override
     @Transactional
-    public Note create(Note note) {
-        Boolean isExistByUserId = restTemplate.getForObject("http://user/api/v1/users/{id}",
+    public final Note create(final Note note) {
+        Boolean isExistByUserId = restTemplate
+                .getForObject("http://user/api/v1/users/{id}",
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java]: @vpetrovaa host is hardcoded, don't use absolute path

File: src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java [hunk 5]
```diff
@@ -17,55 +20,86 @@
...
-            throw new ResourceDoesNotExistException("There are no user with id " + note.getUserId());
+            throw new ResourceDoesNotExistException(
+                    "There are no user with id " + note.getUserId()
+            );
         }
-        noteRepository.save(note);
-        return note;
+
+        NoteEvent noteEvent = new NoteEvent();
+        noteEvent.setType(NoteEvent.Method.POST);
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java:49]: I got why you are using POST

File: src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java [hunk 6]
```diff
@@ -17,55 +20,86 @@
...
-    public Note update(Note note) {
+    public final Note update(final Note note) {
         Note noteFromDb = noteRepository.findById(note.getId()).get();
         if (!noteFromDb.getTheme().equals(note.getTheme())) {
-            throw new IllegalOperationException("You cant change the theme of note");
+            throw new IllegalOperationException(
+                    "You cant change the theme of note"
+            );
         }
+
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java]: @vpetrovaa empty lines it is not a good idea

File: src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java [hunk 7]
```diff
@@ -17,55 +20,86 @@
...
+        noteEvent.setUserId(note.getUserId());
+        kfProducer.sendMessage(noteEvent);
         return noteRepository.save(note);
     }
 
     @Override
-    public void delete(Long id) {
+    public final void delete(final Long id) {
         noteRepository.deleteById(id);
+
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java]: the same story here

File: src/main/java/com/solvd/noteservice/web/dto/util/ConstraintsUtils.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.solvd.noteservice.web.dto.util;
+
+public class ConstraintsUtils {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/web/dto/util/ConstraintsUtils.java:3]: this class is redundant

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java (hunk 6 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java]: @h1alexbel checkstyle checking marked it as error, ignore this case?

File: src/main/java/com/solvd/noteservice/web/dto/util/ConstraintsUtils.java (hunk 8 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/noteservice/web/dto/util/ConstraintsUtils.java:3]: @h1alexbel checkstyle marked number parameter in @Length(20) as error, ignore this case too?

File: src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java (hunk 7 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java]: @h1alexbel https://github.com/vpetrovaa/note-service/pull/15/commits/65e1ec219338ddbd8be431d2c7618e0aced922d6 fixed

File: src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java (hunk 4 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/noteservice/service/impl/NoteServiceImpl.java]: @h1alexbel fixed https://github.com/vpetrovaa/note-service/pull/15/commits/41bc54cbe4c92d6d9f7023fb2ec274b206bb62f9

File: src/main/java/com/solvd/noteservice/web/WebConfig.java [hunk 9]
```diff
@@ -13,7 +13,7 @@ public class WebConfig {
 
     @Bean
     @LoadBalanced
-    final RestTemplate restTemplate() {
+    RestTemplate restTemplate() {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/noteservice/web/WebConfig.java]: @vpetrovaa why this method is marked as `public`?

Comment (vpetrovaa) [reply src/main/java/com/solvd/noteservice/web/WebConfig.java]: @vpetrovaa I got the exception when it was final

File: src/main/java/com/solvd/noteservice/kafka/KfProducer.java (hunk 1 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/noteservice/kafka/KfProducer.java:7]: @h1alexbel in interface - void sendMessage(Object data);
in implementation - public final void sendMessage(final Object noteEvent) { ... send ( (NoteEvent) noteEvent ) ...};
is it a good idea?

File: src/main/java/com/solvd/noteservice/web/WebConfig.java (hunk 9 above)
Comment (h1alexbel) [reply src/main/java/com/solvd/noteservice/web/WebConfig.java]: @vpetrovaa Yes, I know, but I mean why it's not `public` method?

Comment (vpetrovaa) [reply src/main/java/com/solvd/noteservice/web/WebConfig.java]: understand, https://github.com/vpetrovaa/note-service/pull/15/commits/8191b5a5c04e398dd17d031e6545b63b6eb90a32

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**